packages/connect/src/protocol-connect/validate-response.spec.ts

```diff
@@ -15,51 +15,85 @@
 import { MethodKind } from "@bufbuild/protobuf";
 import { validateResponse } from "./validate-response.js";
 import { ConnectError } from "../connect-error.js";
+import { Code } from "../code.js";
 
 describe("Connect validateResponse()", function () {
   describe("with unary", function () {
-    const methodKind = MethodKind.Unary;
-    it("should be successful for HTTP 200", function () {
-      const r = validateResponse(methodKind, 200, new Headers());
+    it("should be successful for HTTP 200 with proper unary JSON content type", function () {
+      const r = validateResponse(
+        MethodKind.Unary,
+        200,
+        new Headers({ "Content-Type": "application/json" })
+      );
       expect(r.isUnaryError).toBeFalse();
       expect(r.unaryError).toBeUndefined();
     });
     it("should return error for HTTP 204", function () {
-      const r = validateResponse(methodKind, 204, new Headers());
+      const r = validateResponse(
+        MethodKind.Unary,
+        204,
+        new Headers({ "Content-Type": "application/json" })
+      );
       expect(r.isUnaryError).toBeTrue();
       expect(r.unaryError?.message).toBe("[unknown] HTTP 204");
     });
-    it("should return error for HTTP error status", function () {
-      const r = validateResponse(methodKind, 400, new Headers());
-      expect(r.isUnaryError).toBeTrue();
-      expect(r.unaryError?.message).toBe("[invalid_argument] HTTP 400");
-    });
     it("should include headers as error metadata", function () {
-      const r = validateResponse(methodKind, 204, new Headers({ Foo: "Bar" }));
+      const r = validateResponse(
+        MethodKind.Unary,
+        204,
+        new Headers({ "Content-Type": "application/json", Foo: "Bar" })
+      );
       expect(r.unaryError?.metadata.get("Foo")).toBe("Bar");
     });
-  });
-  describe("with streaming", function () {
-    const methodKind = MethodKind.ServerStreaming;
-    it("should be successful for HTTP 200", function () {
-      const r = validateResponse(methodKind, 200, new Headers());
+    it("should be successful for HTTP 200 with proper unary proto content type", function () {
+      const r = validateResponse(
+        MethodKind.Unary,
+        200,
+        new Headers({ "Content-Type": "application/proto" })
+      );
       expect(r.isUnaryError).toBeFalse();
       expect(r.unaryError).toBeUndefined();
     });
-    it("should throw error for HTTP error status", function () {
+    it("should throw error for HTTP error status with binary response body", function () {
       try {
-        validateResponse(methodKind, 400, new Headers());
-        fail("expected error");
+        validateResponse(
+          MethodKind.Unary,
+          400,
+          new Headers({
+            "Content-Type": "application/proto",
+          })
+        );
       } catch (e) {
```

Comment:
This test will pass even if validate response never throws. It is missing a `fail`

Reply:
Fixed. Thanks for catching that